### PR TITLE
test: add comprehensive integration and error-path test coverage

### PR DESCRIPTION
**Pull request by eunolabs**
## Summary

Adds 1,000+ lines of targeted integration and error-path tests across the codebase to reach previously untested branches: stdio proxy lifecycle against real subprocesses, audit log rotation edge cases, HTTP drift checking, Redis kill-switch error paths, manifest JSON field validation, and enforcement engine type handling.

## Why

The existing test suite had gaps in branches that are difficult or impossible to reach through public APIs:
- `StdioProxy.Start` full lifecycle (connect → initialize → drift check → serve → drain) requires a real subprocess upstream, not just in-memory mocks
- Audit log rotation rename failures and multi-key file loading need filesystem state manipulation
- HTTP drift checking and Redis kill-switch error paths need selective fake implementations to inject failures at specific points
- Enforcement engine's JSON type discriminator and condition type assertions need white-box testing
- Manifest unknown-field guards need reflective JSON key extraction validation

These gaps left critical fail-closed paths (e.g., drift check abort on description hash mismatch, audit rotation error handling) unverified. The new tests exercise these branches and document expected behavior.

## Type of change

- Test coverage (no behavior change)

## How it was tested

```
make test    # All new tests pass; pkg/ coverage improves from 80% baseline
make lint    # No new findings
```

New tests are organized by concern:
- `stdio_integration_test.go`: Full `StdioProxy` lifecycle, subprocess wiring, HTTP upstream routing, live tool introspection, drift check skip/fatal branches
- `audit_rotate_keys_test.go`: Rotation success, rename failure with open fd, multi-key file loading, malformed key detection
- `doctor_live_test.go`: Manifest loading/merging, audit-only counting, live drift validation with indented output
- `drift_http_test.go`: HTTP drift check clean/skip/fatal branches via pipe-backed session
- `redis_errorpaths_test.go`: Get/SCAN/DEL error injection for refresh, reset, and deleteByPrefix
- `internal_coverage_test.go` (enforcement): JSON type discriminator, schema type compatibility, SequenceBlockCondition type assertion
- `manifest_jsonfieldkeys_test.go`: JSON field key extraction for all tag shapes (named, options, skipped, untagged, embedded)
- `stats_test.go` (modified): Label precedence for histogram rows (typed target, bare target, legacy tool)
- `coverage_test.go` (capability, modified): Obligation unmarshal error cases for redactFields and annotate
- `suggest_test.go` (capability, modified): Empty candidate list edge case

## Security considerations

No security impact. Tests verify existing fail-closed behavior (drift check abort, audit rotation error handling, malformed key rejection) without changing implementation. All new code paths are test-only.

## Checklist

- [x] `make test` passes — race detector clean, `pkg/` coverage improves to >80%
- [x] `make lint` passes — no new findings
- [x] `make check-license` passes — Apache-2.0 header on every new file
- [x] No behavior changes; docs unchanged

https://claude.ai/code/session_012uPpuyA7aigMMvGfLHmwzE

### CODE REVIEW

File: cmd/eunox-mcp/manifest_jsonfieldkeys_test.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+type jfkSample struct {
+	jfkEmbed          // anonymous embed → promotes "inner"
+	Named      string `json:"named"`
+	WithOpts   int    `json:"with_opts,omitempty"`
+	Skipped    string `json:"-"`
+	Untagged   string // no tag → falls back to the field name "Untagged"
+	PtrField   *jfkEmbed
+}
```
Comment (eunolabs) [inline cmd/eunox-mcp/manifest_jsonfieldkeys_test.go:27]: Nit: this file isn't `gofmt`-clean. gofmt re-aligns these struct fields (and the `want` map at L33-39) — the names/types currently carry extra padding. `gofmt -w cmd/eunox-mcp/manifest_jsonfieldkeys_test.go` fixes it.

`make lint` doesn't gate formatting (the golangci v2 config has no `formatters:` section), so CI stays green — but the rest of the tree is gofmt-clean, so this stands out.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: cmd/eunox-mcp/stdio_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,487 @@
...
+// stdin/stdout with pipes) and starts it in the background.
+func newStdioHostHarness(t *testing.T, opts StdioProxyOptions) *stdioHostHarness {
+	t.Helper()
+	cmd, args := helperUpstreamArgs()
+	opts.Command = cmd
+	opts.Args = args
+
+	p := NewStdioProxy(opts)
+
+	inR, inW := io.Pipe()  // host → proxy
```
Comment (eunolabs) [inline cmd/eunox-mcp/stdio_integration_test.go]: Nit: also not `gofmt`-clean — there's a double space before the comment here (`io.Pipe()  //`). `gofmt -w` on this file collapses it to a single space.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: cmd/eunox-mcp/stdio_integration_test.go [hunk 3]
```diff
@@ -0,0 +1,487 @@
...
+	}
+
+	if err := h.shutdown(t); err != nil {
+		t.Errorf("Start returned error: %v", err)
+	}
+}
+
+// TestStdioProxy_StrictDriftAbortsStart asserts that a manifest pinning a tool
+// the upstream does not expose (FM-2: unexpected/missing tool) aborts Start when
+// strict drift is enabled, and the upstream is killed.
```
Comment (eunolabs) [inline cmd/eunox-mcp/stdio_integration_test.go]: The doc comment says the manifest pins "a tool the upstream does not expose (FM-2: unexpected/missing tool)", but the test actually pins `read_file` — which `TestHelperStdioUpstream` *does* expose — with a bogus `descriptionHash`. That's FM-5 description drift, exactly as the inline comment at L275 says. The header contradicts the body and the assertion; suggest dropping the FM-2 framing so the doc reads as an FM-5 drift abort.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: cmd/eunox-mcp/drift_http_test.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+// cannedToolsSession returns an httpSession backed by io.Pipe that answers a
+// single tools/list request with the supplied tool set, then a cleanup func.
+func cannedToolsSession(t *testing.T, serverVersion string, toolNames ...string) (sess *httpSession, cleanup func()) {
+	t.Helper()
+	upR, upW := io.Pipe()
+	sess = &httpSession{
+		pending:               make(map[string]chan rpcMsg),
+		done:                  make(chan struct{}),
+		upWriter:              newMsgWriter(upW),
+		upReader:              newMsgReader(upR),
```
Comment (eunolabs) [inline cmd/eunox-mcp/drift_http_test.go]: Minor clarity: `upReader` is wired to a reader on the same `upR` pipe the helper goroutine below consumes, but `runHTTPDriftCheck` never reads from it on this path — the goroutine reads the request and answers directly via the `pending` channel. Two readers on one pipe reads as a race waiting to happen to the next person in this file (it isn't, since the session never reads `upReader` here). Consider dropping the `upReader` field from this helper since it's unused.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by eunolabs — COMMENTED** [review]
Reviewed this test-coverage PR locally on `claude/ecstatic-hopper-bh2a2z`:

- `go vet ./...` — clean
- All new tests pass under `-race` (`cmd/eunox-mcp`, `pkg/killswitch`, `pkg/enforcement`, `pkg/capability`)
- Apache-2.0 SPDX headers present on every new file

The tests are well-targeted and assert real behavior rather than just touching lines — the fail-closed branches (drift abort on `descriptionHash` mismatch, malformed audit-key → hard error, the sequential Redis SCAN/DEL error branches isolated via the selective `redis.Cmdable` fake) are exactly the spots that were hard to reach through the public API, and they're verified rather than just executed. Re-execing the test binary as a real stdio upstream to drive the full `StdioProxy.Start` lifecycle is a clean way to cover the subprocess wiring.

A few non-blocking nits inline:
- Two new files aren't `gofmt`-clean (`manifest_jsonfieldkeys_test.go`, `stdio_integration_test.go`). CI is green because `make lint` doesn't gate formatting, but the rest of the tree is gofmt-clean — worth a `gofmt -w`.
- One misleading doc comment on `TestStdioProxy_StrictDriftAbortsStart` (says FM-2 / missing tool; the test actually exercises FM-5 description drift).
- One unused `upReader` field in the `drift_http_test.go` session helper.

Note: `cmd/eunox-mcp/stats_test.go` also shows as not gofmt-clean, but that's pre-existing on `main` (lines 52/89), not introduced here.

Nothing blocking — LGTM once the gofmt nits are run through.

---
_Generated by [Claude Code](https://claude.ai/code)_
